File: backend/app/schemas/workload.py
from enum import Enum

from pydantic import BaseModel, Field


class WorkloadType(str, Enum):
    realtime = "realtime"
    batch = "batch"
    flexible = "flexible"
    critical = "critical"
    non_critical = "non-critical"


class WorkloadRequest(BaseModel):
    name: str = Field(
        min_length=2,
        max_length=100,
    )

    cpu_usage: float = Field(
        ge=0,
        le=100,
    )

    memory_usage: float = Field(
        ge=0,
        le=100,
    )

    replicas: int = Field(
        ge=1,
        le=1000,
    )

    runtime_hours: float = Field(
        gt=0,
        le=24,
    )

    requests_per_day: int = Field(
        ge=0,
    )

    workload_type: WorkloadType


class Recommendation(BaseModel):
    code: str
    title: str
    message: str
    severity: str
    estimated_saving_percent: int | None = None


class WorkloadAnalysisResponse(BaseModel):
    workload_name: str
    efficiency_score: int
    status: str
    recommendations: list[Recommendation]

File: frontend/src/types/workload.ts
export type WorkloadType =
  | "realtime"
  | "batch"
  | "flexible"
  | "critical"
  | "non-critical";


export type WorkloadRequest = {
  name: string;
  cpu_usage: number;
  memory_usage: number;
  replicas: number;
  runtime_hours: number;
  requests_per_day: number;
  workload_type: WorkloadType;
};


export type Recommendation = {
  code: string;
  title: string;
  message: string;
  severity: string;
  estimated_saving_percent: number | null;
};


export type WorkloadAnalysis = {
  workload_name: string;
  efficiency_score: number;
  status: string;
  recommendations: Recommendation[];
};